### PR TITLE
[WIP][Doc] Modify Documentation for use with Monorepo

### PR DESCRIPTION
**Pull request by Tirokk**
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-04-18T09:19:54Z" title="Thursday, April 18th 2019, 11:19:54 am +02:00">Apr 18, 2019</time>_
_Closed <time datetime="2019-06-14T19:18:09Z" title="Friday, June 14th 2019, 9:18:09 pm +02:00">Jun 14, 2019</time>_
---

### Issues

- relates #479

### Checklist

- [X] None

### Todo

- [X] None

### CODE REVIEW

**Comment by Tirokk** [comment]
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-04-18T09:28:38Z" title="Thursday, April 18th 2019, 11:28:38 am +02:00">Apr 18, 2019</time>_

---


I used your branch now @roschaefer , because the branches I have created did not show up in GitBook. Some test branches are still lying there to see if they will show up later. I will delete them later …
Now I have seen, that I could have created them in GitBook directly.
It is still work in progress …

Editing in GitBook did not work correctly in Safari. That took me a while to figure out. Now on FireFox is kind of fine.

In new Tab Tables I created are always pre paragraph lines in front of my text, that I can not delete. But we can correct that hopefully later.

**Comment by Tirokk** [comment]
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-04-24T13:07:14Z" title="Wednesday, April 24th 2019, 3:07:14 pm +02:00">Apr 24, 2019</time>_

---


I improved the docs for monorepo with GitBook. First approach.
Please review.

File: SUMMARY.md [hunk 1]
```diff
@@ -1,24 +1,28 @@
 # Table of contents
 
 * [Introduction](README.md)
-* [Edit this Documentation](edit-this-documentation.md)
-* [Installation](installation.md)
+* [Installation](installation/INSTALLATION.md)
+  * [Docker Installation](installation/INSTALLATION-DOCKER.md)
+  * [Local Installation](installation/INSTALLATION-LOCAL.md)
 * [Backend](backend/README.md)
```
Comment (Tirokk) [inline SUMMARY.md:7]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-24T13:22:51Z" title="Wednesday, April 24th 2019, 3:22:51 pm +02:00">Apr 24, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,28 +1,34 @@
 # Table of contents
 
 * [Introduction](README.md)
-* [Edit this Documentation](edit-this-documentation.md)
-* [Installation](installation.md)
+* [Installation](installation/README.md)
+  * [Docker Installation](installation/installation-docker.md)
+  * [Local Installation](installation/installation-local.md)
 * [Backend](backend/README.md)
```


When I visit https://docs.human-connection.org/human-connection/~/revisions/-LdEQ_-L4kwVw-mn1ibR/v/docs-improve_installation_instructions/backend which is the backend overview page, I see a blank page with links to the subpages.

Screenshot (86)

File: SUMMARY.md [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
+  * [Neo4j Database](neo4j/README.md)
```
Comment (Tirokk) [inline SUMMARY.md:8]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-24T13:26:21Z" title="Wednesday, April 24th 2019, 3:26:21 pm +02:00">Apr 24, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,28 +1,34 @@
 # Table of contents
 
 * [Introduction](README.md)
-* [Edit this Documentation](edit-this-documentation.md)
-* [Installation](installation.md)
+* [Installation](installation/README.md)
+  * [Docker Installation](installation/installation-docker.md)
+  * [Local Installation](installation/installation-local.md)
 * [Backend](backend/README.md)
+  * [Neo4j Database](backend/neo4j-database.md)
```


This whole section `neo4j` will be moved to a top level folder in https://github.com/Human-Connection/Human-Connection/pull/485: 
```suggestion
  * [Neo4j Database](neo4j/README.md)
```

File: SUMMARY.md [hunk 3]
```diff
@@ -1,24 +1,28 @@
...
   * [GraphQL](backend/graphql.md)
+  * [Middleware](backend/middleware.md)
+  * [Legacy Migration](backend/db-migration-worker/README.md)
 * [Webapp](webapp/README.md)
+  * [Styleguide](webapp/styleguide.md)
```
Comment (Tirokk) [inline SUMMARY.md:13]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-24T13:27:58Z" title="Wednesday, April 24th 2019, 3:27:58 pm +02:00">Apr 24, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,28 +1,34 @@
 # Table of contents
 
 * [Introduction](README.md)
-* [Edit this Documentation](edit-this-documentation.md)
-* [Installation](installation.md)
+* [Installation](installation/README.md)
+  * [Docker Installation](installation/installation-docker.md)
+  * [Local Installation](installation/installation-local.md)
 * [Backend](backend/README.md)
+  * [Neo4j Database](backend/neo4j-database.md)
   * [GraphQL](backend/graphql.md)
-  * [Legacy Migration](backend/db-migration-worker/README.md)
+  * [Testing](backend/testing.md)
+  * [Middleware](backend/middleware.md)
+  * [Legacy Migration](backend/db-migration-worker.md)
```


Same, legacy migration is completely irrelevant for open source contributors. It's being moved into folder `deployment` in https://github.com/Human-Connection/Human-Connection/pull/485

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-04-24T13:33:27Z" title="Wednesday, April 24th 2019, 3:33:27 pm +02:00">Apr 24, 2019</time>_

---


Oh, and could you `git fetch origin` and `git merge origin/master` @Tirokk ? You have merge conflicts.

**Comment by Tirokk** [comment]
> [codecov-io](https://github.com/codecov-io) **Authored by [codecov-io](https://github.com/codecov-io)**
_<time datetime="2019-04-25T20:13:25Z" title="Thursday, April 25th 2019, 10:13:25 pm +02:00">Apr 25, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/483?src=pr&el=h1) Report
> Merging [#483](https://codecov.io/gh/Human-Connection/Human-Connection/pull/483?src=pr&el=desc) into [master](https://codecov.io/gh/Human-Connection/Human-Connection/commit/3cfa5b0cd4e469e979556a20fee69e2d5a981035?src=pr&el=desc) will **not change** coverage.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/Human-Connection/Human-Connection/pull/483?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #483   +/-   ##
=======================================
  Coverage   18.63%   18.63%           
=======================================
  Files         113      113           
  Lines        1878     1878           
  Branches       57       57           
=======================================
  Hits          350      350           
  Misses       1506     1506           
  Partials       22       22
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/483?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/483?src=pr&el=footer). Last update [3cfa5b0...2b85d57](https://codecov.io/gh/Human-Connection/Human-Connection/pull/483?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-06-12T08:34:49Z" title="Wednesday, June 12th 2019, 10:34:49 am +02:00">Jun 12, 2019</time>_

---


@Tirokk I think this PR got stale. The biggest issue which the PR should address has been already resolved: We have one setup for docker including all services whereas the documentation for local installation  has setup instructions for each service individually.

@Tirokk could you update this PR? Maybe there are still fragments which we can re-use.

**Comment by Tirokk** [comment]
> [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-06-14T19:18:09Z" title="Friday, June 14th 2019, 9:18:09 pm +02:00">Jun 14, 2019</time>_

---


Your right @roschaefer and I close the PR.

**Review by Mogge — CHANGES_REQUESTED** [review]
Great job @Tirokk however I would prefer to have separate installation instructions in each top-level folder. I would not put it into one. Why? People might want to read through the docs on Github, e.g. if you visit [backend/](https://github.com/Human-Connection/Human-Connection/tree/docs-improve_installation_instructions/backend) there is no README. This subfolder should have installation instructions how to setup the backend only.

So we want to have best of both worlds: Documentation on Gitbook (in the right order, via `SUMMARY.md`) but we also want to have the docs residing in the repository on Github.

**Pull request closed without merging**